source/extensions/filters/http/jwt_authn/authenticator.cc

```diff
@@ -88,13 +246,22 @@ void AuthenticatorImpl::verify(Http::HeaderMap& headers, Authenticator::Callback
   callback_ = callback;
 
   ENVOY_LOG(debug, "Jwt authentication starts");
+
+  ASSERT(headers.Path());
+  const auto& ret = tryMatch(headers.Path()->value());
+  if (!ret.first || ret.second->getRequirementVerifier() == nullptr) {
+    ENVOY_LOG(debug, "Path did not match any requires, or requires not set.");
+    doneWithStatus(Status::Ok);
+    return;
+  }
+  JwtVerifierPtr jwtVerifierPtr = ret.second->getRequirementVerifier();
+
+  okToBypass_ =
+      ret.second->isAllowMissingOrFailedVerifier() && Status::Ok == jwtVerifierPtr->verify({});
+
   auto tokens = config_->getExtractor().extract(headers);
```

Comment:
Ideally,  we only need to extract the tokens for required providers.  So Extractor should take a list of required_issuers.